src/d/d_attention.cpp

```diff
@@ -478,47 +733,300 @@ void dAttention_c::runDebugDisp() {
+    }
+
+    mHint.proc();
+    mCatch.proc();
+    mLook[0].proc();
+    mLook[1].proc();
+    return true;
 }
 
 /* 8009F460-8009F5FC       .text Draw__12dAttention_cFv */
 void dAttention_c::Draw() {
     /* Nonmatching */
+    Mtx invCamera;
+    PSMTXInverse(g_dComIfG_gameInfo.drawlist.mpCamera->mViewMtxNoTrans, invCamera);
+    fopAc_ac_c *target = LockonTarget(0);
+    if (g_dComIfG_gameInfo.play.mEvtCtrl.mMode != 0 || g_dComIfG_gameInfo.play.mScopeMesgStatus != 0)
+        return;
+    if (target != NULL) {
+        if (target != NULL) {
+            this->draw[0].draw(target->attention_info.position, invCamera);
+        }
+
+        if (mLockOnNum >= 2 && this->draw[1].mpAnmClr != NULL) {
+            int listIdx = mLockOnOffs;
+
+            if (mLockOnOffs == 0) {
+                listIdx = mLockOnNum - 1;
+            } else {
+                listIdx--;
+            }
+
+            if (mLockOnList[listIdx].getActor() != NULL) {
+                fopAc_ac_c* target = mLockOnList[listIdx].getActor();
+                this->draw[1].draw(target->attention_info.position, invCamera);
+            }
+        }
+
+        fpc_ProcID id = LockonTargetPId(0);
+        this->mlockedOnPId = id;
+        this->mDrawAttnPos = target->attention_info.position;
+        this->field_0x028 = 0;
+    } else {
+        if (this->field_0x028 > 0) {
+            uint temp = this->mlockedOnPId;
+            // unsure about this cast
+            target = reinterpret_cast<fopAc_ac_c*>(fopAcIt_Judge(&fpcSch_JudgeByID, &temp));
+            if (target != NULL) {
+                this->draw[0].draw(target->attention_info.position, invCamera);
+                this->mDrawAttnPos = target->attention_info.position;
+            } else {
+                this->draw[0].draw(this->mDrawAttnPos, invCamera);
+            }
+        }
+    }
 }
 
 /* 8009F5FC-8009F6B4       .text setAnm__10dAttDraw_cFiii */
-void dAttDraw_c::setAnm(int, int, int) {
-    /* Nonmatching */
+void dAttDraw_c::setAnm(int resIdxTransform, int resIdxColor, int loopMode) {
+    J3DAnmTransform *pAnimRes;
+    J3DAnmColor *color;
+
+    pAnimRes = (J3DAnmTransform *) dRes_control_c::getRes("Always", resIdxTransform, g_dComIfG_gameInfo.mResControl.mObjectInfo, 0x40);
+    this->anm->setAnm(pAnimRes, loopMode, 0.0, 1.0, 0.0, -1.0, NULL);
+    if (resIdxColor < 0) {
+        this->mpAnmClr = NULL;
+    }
+    else {
+        color = (J3DAnmColor *)dRes_control_c::getRes("Always", resIdxColor, g_dComIfG_gameInfo.mResControl.mObjectInfo, 0x40);
+        this->mpAnmClr = color;
+    }
 }
 
 /* 8009F6B4-8009F834       .text draw__10dAttDraw_cFR4cXyzPA4_f */
-void dAttDraw_c::draw(cXyz&, Mtx) {
-    /* Nonmatching */
+void dAttDraw_c::draw(cXyz &pos, Mtx mtx) {
+    J3DModel *model = this->anm->mpModel;
+    mDoMtx_stack_c::transS(pos);
+    mDoMtx_stack_c::concat(mtx);
+    PSMTXCopy(mDoMtx_stack_c::now, model->getBaseTRMtx());
```

Comment:
Any time you see PSMTX, it's probably an inline
```suggestion
    model->setBaseTRMtx(mDoMtx_stack_c::get());
```

Reply:
There's only a single usage of `MTX*`: `MTXInverse(dComIfGd_getViewRotMtx(), invCamera);`. It does not seem to come from an inline as debug maps don't mention any functions around the call to `dComIfGd_getViewRotMtx`